README.md

```diff
@@ -80,99 +64,145 @@ Semgrep exists because:
+```
 
-**Example**: Say you want to find all calls to a function named `exec`, and you don't care about the arguments. With Semgrep, you could simply supply the pattern `exec(...)` and you'd match:
+This matches on code like the following:
 
-~~~python
-# Simple cases grep finds
-exec("ls")
+```py
+# Semgrep matches simple cases that grep also finds
 exec(some_var)
 
-# But you don't have to worry about whitespace
+# Semgrep also matches code with whitespace that grep misses
 exec (foo)
 
-# Or calls across multiple lines
+# And Semgrep matches calls across multiple lines that grep also misses
 exec (
     bar
 )
-~~~
+```
+
+This can be taken further with additional syntax, such as:
 
-Importantly, Semgrep would *not* match the following:
+| **Pattern**                                                        | **Match**                                                |
+|:-------------------------------------------------------------------|:-----------------------------------------------------------|
+| [`$X == $X`](https://semgrep.live/20B)                             | `if (node.id == node.id): ...`                             |
+| [`requests.get(..., verify=False, ...)`](https://semgrep.live/jqn) | `requests.get(url, timeout=3, verify=False)`               |
+| [`os.system(...)`](https://semgrep.live/1W5)                       | `from os import system; system('echo semgrep')`            |
+| [`$ELEMENT.innerHTML`](https://semgrep.live/9ze)                   | ``el.innerHTML = "<img src='x' onerror='alert(`XSS`)'>";`` |
+| [`$TOKEN.SignedString([]byte("..."))`](https://semgrep.live/rXW)   | `ss, err := token.SignedString([]byte("HARDCODED KEY"))`   |
 
-~~~python
-# grep would match this, but Semgrep ignores it because
-# it doesn't have the right function name
-other_exec(bar)
 
-# Semgrep ignores commented out lines
-# exec(foo)
+A [multi-pattern YAML syntax]() then lets you combine simple patterns using boolean logic to create powerful rules for continuous scanning:
 
-# and hard-coded strings
-print("exec(bar)")
-~~~
+```yaml
+id: flask-insecure-cookie
+patterns:
+  - pattern: flask.response.set_cookie(...)
+  - pattern-not: flask.response.set_cookie(..., httponly=True, secure=True, samesite='Lax', ...)
+message: |
+  Flask cookies should be handled securely by setting `secure=True`, `httponly=True`, and `samesite='Lax'`.
+```
 
-Semgrep will even match aliased imports:
+See [Installation](#installation) and [Workflows](#workflows) to learn more.
 
-~~~python
-# Semgrep knows that safe_function refers to exec so it
-# will still match!
-#   Oof, try finding this with grep
-import exec as safe_function
-safe_function(tricksy)
-~~~
+## Installation
 
-Play with this example in your browser [here](https://semgrep.live/QrkD), or copy the above code into a file locally (`exec.py`) and run:
+TODO
 
-~~~python
-$ semgrep -l python -e "exec(...)" /path/to/exec.py
-~~~
+## Usage
 
-More example patterns:
+### Command Line Options
```

Comment:
Why would we put this in the README?

Reply:
Historically we've gotten feedback that it's a good way to learn what the tool does and how it works (from Bento days). I added it here just to see how it felt and am not sold, at least until `--help` is refined.